Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -868,19 +868,14 @@ private void showMotionControlsOptions()
             {
               IntSetting.MAIN_MOTION_CONTROLS.setInt(mSettings, indexSelected);
 
-              if (indexSelected != 2)
-                mMotionListener.enable();
-              else
-                mMotionListener.disable();
-
               File wiimoteNewFile = SettingsFile.getSettingsFile(Settings.FILE_WIIMOTE);
               IniFile wiimoteNewIni = new IniFile(wiimoteNewFile);
               wiimoteNewIni.setBoolean("Wiimote1", "IMUIR/Enabled", indexSelected != 1);
               wiimoteNewIni.save(wiimoteNewFile);
 
               NativeLibrary.ReloadWiimoteConfig();
             });
-    builder.setPositiveButton(R.string.ok, (dialogInterface, i) -> dialogInterface.dismiss());
+    builder.setPositiveButton(R.string.ok, (dialogInterface, i) -> updateMotionListener());
```

Comment:
We also need to call `updateMotionListener` in the case where the user dismisses the dialog.

Reply:
Oh right. I should probably just replace the removed code with `updateMotionListener` in that case.